File: src/queue.cpp
#include "queue.h"

PacketQueue Queue;

PacketQueue::PacketQueue() {
    head = 0;
    tail = 0;
    count = 0;
}

bool PacketQueue::push(const Packet& packet) {
    if (count == QUEUE_SIZE)
        return false;

    queue[tail] = packet;
    tail = (tail + 1) % QUEUE_SIZE;
    count++;
    return true;
}

bool PacketQueue::pop(Packet& packet) {
    if (count == 0)
        return false;

    packet = queue[head];
    head = (head + 1) % QUEUE_SIZE;
    count--;
    return true;
}

bool PacketQueue::empty() const {
    return count == 0;
}

bool PacketQueue::full() const {
    return count == QUEUE_SIZE;
}

uint8_t PacketQueue::size() const {
    return count;
}

File: deamon/queue.py
from queue import Queue

class commandQueue:
    def __init__(self):
        self.queue = Queue()

    def push(self, command):
        self.queue.put(command)

    def pop(self):
        if self.queue.empty():
            return None
        return self.queue.get()

    def empty(self):
        return self.queue.empty()
